src/providers/WorkflowCore.Persistence.EntityFramework/Services/EntityFrameworkPersistenceProvider.cs

```diff
@@ -29,10 +29,11 @@ public async Task<string> CreateEventSubscription(EventSubscription subscription
         {
             using (var db = ConstructDbContext())
             {
-                subscription.Id = Guid.NewGuid().ToString();
+                subscription.Id = Guid.Empty.ToString();
                 var persistable = subscription.ToPersistable();
                 var result = db.Set<PersistedSubscription>().Add(persistable);
                 await db.SaveChangesAsync(cancellationToken);
+                subscription.Id = result.Entity.SubscriptionId.ToString();
```

Comment:
return result.Entity.SubscriptionId.ToString()

actually no need for this re-assign I guess

Reply:
I'd like to leave it for safety, e.g. someone makes a change later and will not know this hack is needed.

Reply:
I would leave it as well.